// @angular/cdk/typings/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/cdk';
import * as i2 from '@angular/common';
import * as i3 from '@angular/platform-browser';
export const A11yModuleNgFactory:i0.NgModuleFactory<i1.A11yModule> = i0.ɵcmf(i1.A11yModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(4608,
          i2.NgLocalization,i2.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(4608,i1.Platform,
          i1.Platform,([] as any[])),i0.ɵmpd(4608,i1.InteractivityChecker,i1.InteractivityChecker,
          [i1.Platform]),i0.ɵmpd(4608,i1.FocusTrapFactory,i1.FocusTrapFactory,[i1.InteractivityChecker,
          i1.Platform,i0.NgZone]),i0.ɵmpd(5120,i1.LiveAnnouncer,i1.LIVE_ANNOUNCER_PROVIDER_FACTORY,
          [[3,i1.LiveAnnouncer],[2,i1.LIVE_ANNOUNCER_ELEMENT_TOKEN],i1.Platform]),
          i0.ɵmpd(512,i2.CommonModule,i2.CommonModule,([] as any[])),i0.ɵmpd(512,i1.PlatformModule,
              i1.PlatformModule,([] as any[])),i0.ɵmpd(512,i1.A11yModule,i1.A11yModule,
              ([] as any[]))]);
    });
export const PlatformModuleNgFactory:i0.NgModuleFactory<i1.PlatformModule> = i0.ɵcmf(i1.PlatformModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(4608,
          i1.Platform,i1.Platform,([] as any[])),i0.ɵmpd(512,i1.PlatformModule,i1.PlatformModule,
          ([] as any[]))]);
    });
export const BidiModuleNgFactory:i0.NgModuleFactory<i1.BidiModule> = i0.ɵcmf(i1.BidiModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(6144,
          i1.DIR_DOCUMENT,(null as any),[i3.DOCUMENT]),i0.ɵmpd(4608,i1.Directionality,
          i1.Directionality,[[2,i1.DIR_DOCUMENT]]),i0.ɵmpd(512,i1.BidiModule,i1.BidiModule,
          ([] as any[]))]);
    });
export const CdkTableModuleNgFactory:i0.NgModuleFactory<i1.CdkTableModule> = i0.ɵcmf(i1.CdkTableModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(4608,
          i2.NgLocalization,i2.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(512,i2.CommonModule,
          i2.CommonModule,([] as any[])),i0.ɵmpd(512,i1.CdkTableModule,i1.CdkTableModule,
          ([] as any[]))]);
    });
export const PortalModuleNgFactory:i0.NgModuleFactory<i1.PortalModule> = i0.ɵcmf(i1.PortalModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(512,
          i1.PortalModule,i1.PortalModule,([] as any[]))]);
    });
export const ObserveContentModuleNgFactory:i0.NgModuleFactory<i1.ObserveContentModule> = i0.ɵcmf(i1.ObserveContentModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(4608,
          i1.MdMutationObserverFactory,i1.MdMutationObserverFactory,([] as any[])),
          i0.ɵmpd(512,i1.ObserveContentModule,i1.ObserveContentModule,([] as any[]))]);
    });
const styles_CdkTable:any[] = ([] as any[]);
export const RenderType_CdkTable:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_CdkTable,
    data:{}});
export function View_CdkTable_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵqud(402653184,1,{_rowPlaceholder:0}),i0.ɵqud(402653184,2,{_headerRowPlaceholder:0}),
      (_l()(),i0.ɵted((null as any),['\n  '])),(_l()(),i0.ɵeld(16777216,(null as any),
          (null as any),1,(null as any),(null as any),(null as any),(null as any),
          (null as any),(null as any),(null as any))),i0.ɵdid(16384,[[2,4]],0,i1.HeaderRowPlaceholder,
          [i0.ViewContainerRef],(null as any),(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n  '])),(_l()(),i0.ɵeld(16777216,(null as any),(null as any),1,(null as any),
          (null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(16384,[[1,4]],0,i1.RowPlaceholder,[i0.ViewContainerRef],(null as any),
          (null as any))],(null as any),(null as any));
}
export function View_CdkTable_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'cdk-table',[['class',
      'cdk-table']],(null as any),(null as any),(null as any),View_CdkTable_0,RenderType_CdkTable)),
      i0.ɵdid(5750784,(null as any),3,i1.CdkTable,[i0.IterableDiffers,i0.ChangeDetectorRef,
          i0.ElementRef,i0.Renderer2,[8,(null as any)]],(null as any),(null as any)),
      i0.ɵqud(603979776,1,{_columnDefinitions:1}),i0.ɵqud(335544320,2,{_headerDefinition:0}),
      i0.ɵqud(603979776,3,{_rowDefinitions:1})],(_ck,_v) => {
    _ck(_v,1,0);
  },(null as any));
}
export const CdkTableNgFactory:i0.ComponentFactory<i1.CdkTable<any>> = i0.ɵccf('cdk-table',
    i1.CdkTable,View_CdkTable_Host_0,{trackBy:'trackBy',dataSource:'dataSource'},{},
    ([] as any[]));
const styles_CdkRow:any[] = ([] as any[]);
export const RenderType_CdkRow:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_CdkRow,
    data:{}});
export function View_CdkRow_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵeld(16777216,(null as any),(null as any),1,(null as any),
      (null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(16384,(null as any),0,i1.CdkCellOutlet,[i0.ViewContainerRef],(null as any),
          (null as any))],(null as any),(null as any));
}
export function View_CdkRow_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'cdk-row',[['class',
      'cdk-row'],['role','row']],(null as any),(null as any),(null as any),View_CdkRow_0,
      RenderType_CdkRow)),i0.ɵdid(49152,(null as any),0,i1.CdkRow,([] as any[]),(null as any),
      (null as any))],(null as any),(null as any));
}
export const CdkRowNgFactory:i0.ComponentFactory<i1.CdkRow> = i0.ɵccf('cdk-row',i1.CdkRow,
    View_CdkRow_Host_0,{},{},([] as any[]));
const styles_CdkHeaderRow:any[] = ([] as any[]);
export const RenderType_CdkHeaderRow:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_CdkHeaderRow,
    data:{}});
export function View_CdkHeaderRow_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵeld(16777216,(null as any),(null as any),1,(null as any),
      (null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(16384,(null as any),0,i1.CdkCellOutlet,[i0.ViewContainerRef],(null as any),
          (null as any))],(null as any),(null as any));
}
export function View_CdkHeaderRow_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'cdk-header-row',
      [['class','cdk-header-row'],['role','row']],(null as any),(null as any),(null as any),
      View_CdkHeaderRow_0,RenderType_CdkHeaderRow)),i0.ɵdid(49152,(null as any),0,
      i1.CdkHeaderRow,([] as any[]),(null as any),(null as any))],(null as any),(null as any));
}
export const CdkHeaderRowNgFactory:i0.ComponentFactory<i1.CdkHeaderRow> = i0.ɵccf('cdk-header-row',
    i1.CdkHeaderRow,View_CdkHeaderRow_Host_0,{},{},([] as any[]));
