// @uirouter/angular/lib/directives/uiView.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from '@uirouter/angular/lib/directives/uiView';
import * as i3 from '@uirouter/core/lib/router';
const styles_UIView:any[] = ([] as any[]);
export const RenderType_UIView:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_UIView,
    data:{}});
function View_UIView_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵand(0,(null as any),(null as any),0))],(null as any),
      (null as any));
}
function View_UIView_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵncd((null as any),0),(_l()(),i0.ɵand(0,(null as any),(null as any),
      0))],(null as any),(null as any));
}
export function View_UIView_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵqud(402653184,1,{componentTarget:0}),(_l()(),i0.ɵted((null as any),
      ['\n    '])),(_l()(),i0.ɵand(16777216,[[1,3],['componentTarget',2]],(null as any),
      0,(null as any),View_UIView_1)),(_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),
      i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_UIView_2)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n  ']))],(_ck,_v) => {
    var _co:i2.UIView = _v.component;
    const currVal_0:boolean = !_co.componentRef;
    _ck(_v,5,0,currVal_0);
  },(null as any));
}
export function View_UIView_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(16777216,(null as any),(null as any),1,'ui-view',
      ([] as any[]),(null as any),(null as any),(null as any),View_UIView_0,RenderType_UIView)),
      i0.ɵdid(245760,(null as any),0,i2.UIView,[i3.UIRouter,'UIView.PARENT_INJECT',
          i0.ViewContainerRef],(null as any),(null as any))],(_ck,_v) => {
    _ck(_v,1,0);
  },(null as any));
}
export const UIViewNgFactory:i0.ComponentFactory<i2.UIView> = i0.ɵccf('ui-view, [ui-view]',
    i2.UIView,View_UIView_Host_0,{name:'name',_name:'ui-view'},{},['*']);
