// ngx-bootstrap/datepicker/daypicker.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from 'ngx-bootstrap/datepicker/daypicker.component';
import * as i3 from 'ngx-bootstrap/datepicker/datepicker-inner.component';
const styles_DayPickerComponent:any[] = ([] as any[]);
export const RenderType_DayPickerComponent:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_DayPickerComponent,data:{}});
function View_DayPickerComponent_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'button',[['class',
      'btn btn-default btn-secondary btn-sm pull-left'],['tabindex','-1'],['type',
      'button']],(null as any),[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.datePicker.move((0 - 1))) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n          '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'i',[['class','glyphicon glyphicon-chevron-left']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n        ']))],(null as any),(null as any));
}
function View_DayPickerComponent_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'button',[['class',
      'btn btn-default btn-secondary btn-sm pull-left'],['tabindex','-1'],['type',
      'button']],(null as any),[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.datePicker.move((0 - 1))) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['<\n        ']))],
      (null as any),(null as any));
}
function View_DayPickerComponent_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'button',[['class',
      'btn btn-default btn-secondary btn-sm pull-right'],['tabindex','-1'],['type',
      'button']],(null as any),[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.datePicker.move(1)) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n          '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'i',[['class','glyphicon glyphicon-chevron-right']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n        ']))],(null as any),(null as any));
}
function View_DayPickerComponent_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'button',[['class',
      'btn btn-default btn-secondary btn-sm pull-right'],['tabindex','-1'],['type',
      'button']],(null as any),[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.datePicker.move(1)) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['>\n        ']))],
      (null as any),(null as any));
}
function View_DayPickerComponent_6(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'th',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any)))],(null as any),
      (null as any));
}
function View_DayPickerComponent_7(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),5,'th',[['class',
      'text-center']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),2,'small',[['aria-label','labelz.full']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'b',([] as any[]),(null as any),(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['',''])),(_l()(),
          i0.ɵted((null as any),['\n      ']))],(null as any),(_ck,_v) => {
    const currVal_0:any = _v.context.$implicit.abbr;
    _ck(_v,4,0,currVal_0);
  });
}
function View_DayPickerComponent_10(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'td',[['class',
      'text-center']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n          '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'em',([] as any[]),(null as any),(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['',''])),(_l()(),
          i0.ɵted((null as any),['\n        ']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.weekNumbers[(<any>(<any>_v.parent).parent).context.index];
    _ck(_v,3,0,currVal_0);
  });
}
function View_DayPickerComponent_12(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),8,'button',[['style',
      'min-width:100%;'],['tabindex','-1'],['type','button']],[[8,'disabled',0]],[[(null as any),
      'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.datePicker.select((<any>_v.parent).context.$implicit.date)) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),i0.ɵpod(['btn-secondary','btn-info','disabled','active',
      'btn-default']),(_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),3,'span',([] as any[]),(null as any),(null as any),
      (null as any),(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,
      [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngClass:[0,
          'ngClass']},(null as any)),i0.ɵpod(['text-muted','text-info']),(_l()(),i0.ɵted((null as any),
      ['',''])),(_l()(),i0.ɵted((null as any),['\n          ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_1:any = i0.ɵinlineInterpolate(1,'btn btn-sm ',(<any>_v.parent).context.$implicit.customClass,
        '');
    const currVal_2:any = _ck(_v,2,0,((_co.isBs4 && !(<any>_v.parent).context.$implicit.selected) && !_co.datePicker.isActive((<any>_v.parent).context.$implicit)),
        (<any>_v.parent).context.$implicit.selected,(<any>_v.parent).context.$implicit.disabled,
        (!_co.isBs4 && _co.datePicker.isActive((<any>_v.parent).context.$implicit)),
        !_co.isBs4);
    _ck(_v,1,0,currVal_1,currVal_2);
    const currVal_3:any = _ck(_v,6,0,((<any>_v.parent).context.$implicit.secondary || (<any>_v.parent).context.$implicit.current),
        (!_co.isBs4 && (<any>_v.parent).context.$implicit.current));
    _ck(_v,5,0,currVal_3);
  },(_ck,_v) => {
    const currVal_0:any = (<any>_v.parent).context.$implicit.disabled;
    _ck(_v,0,0,currVal_0);
    const currVal_4:any = (<any>_v.parent).context.$implicit.label;
    _ck(_v,7,0,currVal_4);
  });
}
function View_DayPickerComponent_11(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'td',[['class',
      'text-center'],['role','gridcell']],[[8,'id',0]],(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n          '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_DayPickerComponent_12)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n        ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_1:boolean = !(_co.datePicker.onlyCurrentMonth && _v.context.$implicit.secondary);
        _ck(_v,3,0,currVal_1);
      },(_ck,_v) => {
        const currVal_0:any = _v.context.$implicit.uid;
        _ck(_v,0,0,currVal_0);
      });
}
function View_DayPickerComponent_9(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),7,'tr',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵand(16777216,(null as any),
      (null as any),1,(null as any),View_DayPickerComponent_10)),i0.ɵdid(16384,(null as any),
      0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_DayPickerComponent_11)),i0.ɵdid(802816,
          (null as any),0,i1.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],
          {ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n      ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.datePicker.showWeeks;
        _ck(_v,3,0,currVal_0);
        const currVal_1:any = (<any>_v.parent).context.$implicit;
        _ck(_v,6,0,currVal_1);
      },(null as any));
}
function View_DayPickerComponent_8(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_DayPickerComponent_9)),i0.ɵdid(16384,
      (null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
      (null as any)),(_l()(),i0.ɵted((null as any),['\n    ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:boolean = !((_co.datePicker.onlyCurrentMonth && _v.context.$implicit[0].secondary) && _v.context.$implicit[6].secondary);
    _ck(_v,2,0,currVal_0);
  },(null as any));
}
function View_DayPickerComponent_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),51,'table',[['aria-activedescendant',
      'activeDateId'],['role','grid']],[[1,'aria-labelledby',0]],(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n  '])),(_l()(),
      i0.ɵeld(0,(null as any),(null as any),42,'thead',([] as any[]),(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
      ['\n    '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),30,'tr',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted((null as any),['\n      '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
      7,'th',([] as any[]),(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_DayPickerComponent_2)),i0.ɵdid(16384,
      (null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
      (null as any)),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_DayPickerComponent_3)),i0.ɵdid(16384,
      (null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
      (null as any)),(_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),i0.ɵted((null as any),
      ['\n      '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),9,'th',([] as any[]),
      [[1,'colspan',0]],(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),6,'button',[['class','btn btn-default btn-secondary btn-sm'],
              ['style','width:100%;'],['tabindex','-1'],['type','button']],[[8,'id',
              0],[8,'disabled',0]],[[(null as any),'click']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:any = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>_co.datePicker.toggleMode()) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,
          [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,
              'klass'],ngClass:[1,'ngClass']},(null as any)),i0.ɵpod(['disabled']),
      (_l()(),i0.ɵted((null as any),['\n          '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'strong',([] as any[]),(null as any),(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['',''])),(_l()(),
          i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵted((null as any),['\n      '])),
      (_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),7,'th',([] as any[]),(null as any),(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_DayPickerComponent_4)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_DayPickerComponent_5)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),
          i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵted((null as any),['\n    '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),7,'tr',([] as any[]),(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n      '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
          View_DayPickerComponent_6)),i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,
          i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n      '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
          View_DayPickerComponent_7)),i0.ɵdid(802816,(null as any),0,i1.NgForOf,[i0.ViewContainerRef,
          i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵted((null as any),['\n  '])),
      (_l()(),i0.ɵted((null as any),['\n  '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          4,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),
          (null as any))),(_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_DayPickerComponent_8)),
      i0.ɵdid(802816,(null as any),0,i1.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n  '])),(_l()(),i0.ɵted((null as any),['\n']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_1:boolean = !_co.isBs4;
    _ck(_v,9,0,currVal_1);
    const currVal_2:any = _co.isBs4;
    _ck(_v,12,0,currVal_2);
    const currVal_6:any = 'btn btn-default btn-secondary btn-sm';
    const currVal_7:any = _ck(_v,19,0,(_co.datePicker.datepickerMode === _co.datePicker.maxMode));
    _ck(_v,18,0,currVal_6,currVal_7);
    const currVal_9:boolean = !_co.isBs4;
    _ck(_v,29,0,currVal_9);
    const currVal_10:any = _co.isBs4;
    _ck(_v,32,0,currVal_10);
    const currVal_11:any = _co.datePicker.showWeeks;
    _ck(_v,39,0,currVal_11);
    const currVal_12:any = _co.labels;
    _ck(_v,42,0,currVal_12);
    const currVal_13:any = _co.rows;
    _ck(_v,49,0,currVal_13);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = (_co.datePicker.uniqueId + '-title');
    _ck(_v,0,0,currVal_0);
    const currVal_3:any = (5 + (_co.datePicker.showWeeks? 1: 0));
    _ck(_v,15,0,currVal_3);
    const currVal_4:any = (_co.datePicker.uniqueId + '-title');
    const currVal_5:any = (_co.datePicker.datepickerMode === _co.datePicker.maxMode);
    _ck(_v,17,0,currVal_4,currVal_5);
    const currVal_8:any = _co.title;
    _ck(_v,22,0,currVal_8);
  });
}
export function View_DayPickerComponent_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n'])),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_DayPickerComponent_1)),i0.ɵdid(16384,
      (null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
      (null as any)),(_l()(),i0.ɵted((null as any),['\n  ']))],(_ck,_v) => {
    var _co:i2.DayPickerComponent = _v.component;
    const currVal_0:any = (_co.datePicker.datepickerMode === 'day');
    _ck(_v,2,0,currVal_0);
  },(null as any));
}
export function View_DayPickerComponent_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'daypicker',([] as any[]),
      (null as any),(null as any),(null as any),View_DayPickerComponent_0,RenderType_DayPickerComponent)),
      i0.ɵdid(114688,(null as any),0,i2.DayPickerComponent,[i3.DatePickerInnerComponent],
          (null as any),(null as any))],(_ck,_v) => {
    _ck(_v,1,0);
  },(null as any));
}
export const DayPickerComponentNgFactory:i0.ComponentFactory<i2.DayPickerComponent> = i0.ɵccf('daypicker',
    i2.DayPickerComponent,View_DayPickerComponent_Host_0,{},{},([] as any[]));
