// ngx-bootstrap/datepicker/yearpicker.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from 'ngx-bootstrap/datepicker/yearpicker.component';
import * as i3 from 'ngx-bootstrap/datepicker/datepicker-inner.component';
const styles_YearPickerComponent:any[] = ([] as any[]);
export const RenderType_YearPickerComponent:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_YearPickerComponent,data:{}});
function View_YearPickerComponent_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),11,'td',[['class',
      'text-center'],['role','gridcell']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),8,'button',[['class','btn btn-default'],
          ['style','min-width:100%;'],['tabindex','-1'],['type','button']],[[8,'disabled',
          0]],[[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.datePicker.select(_v.context.$implicit.date)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,[i0.IterableDiffers,
          i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
          'ngClass']},(null as any)),i0.ɵpod(['btn-link','btn-info','disabled','active']),
      (_l()(),i0.ɵted((null as any),['\n          '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),3,'span',([] as any[]),(null as any),(null as any),(null as any),
          (null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,
          [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngClass:[0,
              'ngClass']},(null as any)),i0.ɵpod(['text-success','text-info']),(_l()(),
          i0.ɵted((null as any),['',''])),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵted((null as any),['\n      ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_1:any = 'btn btn-default';
    const currVal_2:any = _ck(_v,4,0,((_co.isBs4 && !_v.context.$implicit.selected) && !_co.datePicker.isActive(_v.context.$implicit)),
        (_v.context.$implicit.selected || ((_co.isBs4 && !_v.context.$implicit.selected) && _co.datePicker.isActive(_v.context.$implicit))),
        _v.context.$implicit.disabled,(!_co.isBs4 && _co.datePicker.isActive(_v.context.$implicit)));
    _ck(_v,3,0,currVal_1,currVal_2);
    const currVal_3:any = _ck(_v,8,0,(_co.isBs4 && _v.context.$implicit.current),(!_co.isBs4 && _v.context.$implicit.current));
    _ck(_v,7,0,currVal_3);
  },(_ck,_v) => {
    const currVal_0:any = _v.context.$implicit.disabled;
    _ck(_v,2,0,currVal_0);
    const currVal_4:any = _v.context.$implicit.label;
    _ck(_v,9,0,currVal_4);
  });
}
function View_YearPickerComponent_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'tr',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted((null as any),['\n      '])),(_l()(),i0.ɵand(16777216,(null as any),
      (null as any),1,(null as any),View_YearPickerComponent_3)),i0.ɵdid(802816,(null as any),
      0,i1.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,
          'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n    ']))],(_ck,
      _v) => {
    const currVal_0:any = _v.context.$implicit;
    _ck(_v,3,0,currVal_0);
  },(null as any));
}
function View_YearPickerComponent_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),40,'table',[['role',
      'grid']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n  '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          31,'thead',([] as any[]),(null as any),(null as any),(null as any),(null as any),
          (null as any))),(_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),28,'tr',([] as any[]),(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n      '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),6,'th',([] as any[]),
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),3,'button',[['class','btn btn-default btn-sm pull-left'],['tabindex',
              '-1'],['type','button']],(null as any),[[(null as any),'click']],(_v,
              en,$event) => {
            var ad:boolean = true;
            var _co:any = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>_co.datePicker.move((0 - 1))) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n          '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'i',[['class','glyphicon glyphicon-chevron-left']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵted((null as any),
          ['\n      '])),(_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),9,'th',([] as any[]),[[1,'colspan',0]],(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n        '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),6,'button',
          [['class','btn btn-default btn-sm'],['role','heading'],['style','width:100%;'],
              ['tabindex','-1'],['type','button']],[[8,'id',0],[8,'disabled',0]],[[(null as any),
              'click']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:any = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>_co.datePicker.toggleMode()) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,
          [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,
              'klass'],ngClass:[1,'ngClass']},(null as any)),i0.ɵpod(['disabled']),
      (_l()(),i0.ɵted((null as any),['\n          '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'strong',([] as any[]),(null as any),(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['',''])),(_l()(),
          i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵted((null as any),['\n      '])),
      (_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),6,'th',([] as any[]),(null as any),(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),3,'button',[['class','btn btn-default btn-sm pull-right'],
          ['tabindex','-1'],['type','button']],(null as any),[[(null as any),'click']],
          (_v,en,$event) => {
            var ad:boolean = true;
            var _co:any = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>_co.datePicker.move(1)) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n          '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'i',[['class','glyphicon glyphicon-chevron-right']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵted((null as any),
          ['\n      '])),(_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵted((null as any),
          ['\n  '])),(_l()(),i0.ɵted((null as any),['\n  '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),4,'tbody',([] as any[]),(null as any),(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n    '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_YearPickerComponent_2)),
      i0.ɵdid(802816,(null as any),0,i1.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n  '])),(_l()(),i0.ɵted((null as any),['\n']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_3:any = 'btn btn-default btn-sm';
    const currVal_4:any = _ck(_v,18,0,(_co.datePicker.datepickerMode === _co.datePicker.maxMode));
    _ck(_v,17,0,currVal_3,currVal_4);
    const currVal_6:any = _co.rows;
    _ck(_v,38,0,currVal_6);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = (((_co.datePicker.yearColLimit - 2) <= 0)? 1: (_co.datePicker.yearColLimit - 2));
    _ck(_v,14,0,currVal_0);
    const currVal_1:any = (_co.datePicker.uniqueId + '-title');
    const currVal_2:any = (_co.datePicker.datepickerMode === _co.datePicker.maxMode);
    _ck(_v,16,0,currVal_1,currVal_2);
    const currVal_5:any = _co.title;
    _ck(_v,21,0,currVal_5);
  });
}
export function View_YearPickerComponent_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n'])),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_YearPickerComponent_1)),i0.ɵdid(16384,
      (null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
      (null as any)),(_l()(),i0.ɵted((null as any),['\n  ']))],(_ck,_v) => {
    var _co:i2.YearPickerComponent = _v.component;
    const currVal_0:any = (_co.datePicker.datepickerMode === 'year');
    _ck(_v,2,0,currVal_0);
  },(null as any));
}
export function View_YearPickerComponent_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'yearpicker',([] as any[]),
      (null as any),(null as any),(null as any),View_YearPickerComponent_0,RenderType_YearPickerComponent)),
      i0.ɵdid(114688,(null as any),0,i2.YearPickerComponent,[i3.DatePickerInnerComponent],
          (null as any),(null as any))],(_ck,_v) => {
    _ck(_v,1,0);
  },(null as any));
}
export const YearPickerComponentNgFactory:i0.ComponentFactory<i2.YearPickerComponent> = i0.ɵccf('yearpicker',
    i2.YearPickerComponent,View_YearPickerComponent_Host_0,{},{},([] as any[]));
